src/components/Charts/ChartZones.tsx

```diff
@@ -0,0 +1,211 @@
+  marginLeft = 40,
+  marginRight = 5,
+  capY,
+  getTooltipText,
+}: {
+  width: number;
+  height: number;
+  columnData: Point[];
+  zones: LevelInfoMap;
+  capY: number;
+  getTooltipText: (valueY: number) => string;
+  marginTop?: number;
+  marginBottom?: number;
+  marginLeft?: number;
+  marginRight?: number;
+}) => {
+  const chartWidth = width - marginLeft - marginRight;
+  const chartHeight = height - marginTop - marginBottom;
+
+  const data: Point[] = columnData.filter(hasData);
+  const dates: Date[] = columnData.map(getDate).filter(isDate);
+
+  const minDate = d3min(dates) || new Date('2020-01-01');
+  const maxDate = moment().add(2, 'weeks').toDate();
+
+  const xScale = scaleTime({
+    domain: [minDate, maxDate],
+    range: [0, chartWidth],
+  });
+
+  const yDataMin = 0;
+  const yDataMax = d3max(data, getY) || 1;
+  const [yAxisMin, yAxisMax] = getAxisLimits(yDataMin, yDataMax, zones);
+  const yMax = Math.min(capY, yAxisMax);
+
+  const yScale = scaleLinear({
+    domain: [yAxisMin, yMax],
+    range: [chartHeight, 0],
+  });
+
+  const getXCoord = (d: Point) => xScale(getDate(d));
+  const getYCoord = (d: Point) => yScale(Math.min(getY(d), capY));
+
+  const regions = getChartRegions(yAxisMin, yMax, zones);
+  const yTicks = computeTickPositions(yAxisMin, yMax, zones);
+
+  const lastPoint = last(data);
+  const lastPointY = getY(lastPoint);
+  const lastPointZone = getZoneByValue(lastPointY, zones);
+
+  const renderMarker = (d: Point) => (
+    <Style.CircleMarker
+      cx={getXCoord(d)}
+      cy={getYCoord(d)}
+      r={6}
+      fill={getZoneByValue(getY(d), zones).color}
+    />
+  );
+
+  const renderTooltip = (d: Point) => (
+    <Tooltip
+      top={marginTop + getYCoord(d)}
+      left={marginLeft + getXCoord(d)}
+      date={getDate(d)}
+      text={getTooltipText(getY(d))}
+    />
+  );
+
+  return (
+    <ChartContainer<Point>
+      data={data}
+      x={getXCoord}
+      y={getYCoord}
+      renderMarker={renderMarker}
+      renderTooltip={renderTooltip}
+      width={width}
+      height={height}
+      marginTop={marginTop}
+      marginBottom={marginBottom}
+      marginLeft={marginLeft}
+      marginRight={marginRight}
+    >
+      <RectClipGroup width={chartWidth} height={chartHeight}>
+        {regions.map((region, i) => (
+          <Group key={`chart-region-${i}`}>
+            <Style.SeriesLine stroke={region.color}>
+              <ZoneLinePath<Point>
+                data={data}
+                x={getXCoord}
+                y={getYCoord}
+                region={region}
+                width={chartWidth}
+                yScale={yScale}
+                curve={curveLinear}
+              />
+            </Style.SeriesLine>
+            <ZoneAnnotation
+              color={region.color}
+              name={region.name}
+              isActive={lastPointZone.name === region.name}
+              x={xScale(maxDate) - 10}
+              y={yScale(0.5 * (region.valueFrom + region.valueTo))}
```

Comment:
why .5 here?

Reply:
It's the midpoint between `valueFrom` and `valueTo`, maybe `(valueFrom + valueTo) / 2` would be more clear?